core/src/Streamly/Internal/Data/Parser/ParserD/Type.hs

```diff
@@ -378,35 +374,26 @@ instance Functor (Step s) where
     {-# INLINE fmap #-}
     fmap = second
 
-assertStepCount :: Int -> m (Step s b) -> (Step s b)
+{-# INLINE assertStepCount #-}
+assertStepCount :: Int -> Step s b -> Step s b
 assertStepCount i step =
     case step of
         Partial n _ -> assert (i == n) step
         Continue n _ -> assert (i == n) step
         Done n _ -> assert (i == n) step
         Error _ -> step
 
--- XXX Use Int -> s -> m (Step s1 b) to take n into account?
-
 -- | Map an extract function over the state of Step
 --
 {-# INLINE extractStep #-}
 extractStep :: Monad m => (s -> m (Step s1 b)) -> Step s b -> m (Step s1 b)
 extractStep f res =
     case res of
-        -- XXX When step says (Partial n) extract must say (Done n)
-        -- We can assert that.
         Partial n s1 -> assertStepCount n <$> f s1
         Done n b -> return $ assertStepCount n $ Done n b
-        -- XXX What to do with n?
```

Comment:
You can remove `assertStepCount` in the above line.

Reply:
Ah, yes. That's redundant.